### ISSUE TITLE
clang diagnostic "missing-designated-field-initializers" shouldn't fire for default constructed types

### ISSUE THREAD

**Issue by DanielJump** [llvm/llvm-project#147582]
https://godbo.lt/z/6a9hvzb7j

I would only expect this warning to fire for types that don't have an explicit initializer or default constructor.

```
#include <string>

struct T
{
    std::string a;
    std::string b;
};

int main(int, char**)
{
    T t { .a = "hello" };
    return 0;
}
```

**Comment by DanielJump** [comment]
Thanks! Search didn't find that for me :|
